src/transformers/models/bert/modeling_tf_bert.py

```diff
@@ -372,7 +272,11 @@ def call(
         if head_mask is not None:
             attention_probs = tf.multiply(attention_probs, head_mask)
```

Comment:
In the continuity of @sgugger comments on `transpose_for_score` do we really need to split the dimension if we are not going to mask some heads? 

I wonder if the calls to `transpose_for_scores` cannot be moved only in this if branch to avoid unecessary reshaping operations on the common path.

Reply:
All the reshape can be even entirely removed if we introduce our own way to compute a 3D dense. This should not be too difficult.